### DISCUSSION TITLE
Using docker upgrading vaultwarden server looses master password

### DISCUSSION THREAD

**Discussion by Section1977** [dani-garcia/vaultwarden#2310] [Q&A]
please fix or give a tip how to fix
persistentce data is the /data directory
master password is not valid anymore after recreating docker container

**Comment by BlackDex** [comment]
I doubt that this is a bug. I rather think the persistent data folder is incorrect. Please verify if there are still users available via the /admin interface.

**Comment by Section1977** [comment]
I think it is.

I install everything then it works, users are shown in admin panel.

after removing and recreating docker container no user anymore.

please check it

**Comment by BlackDex** [comment]
Well, this is not an issue with Vaultwarden but, as i mentioned before, it looks like the volume is not correctly linked.

**Comment by Section1977** [comment]
ok, my volume is bind to /data within vaultwarden/server:latest
what can i do ?

**Reply by BlackDex** [reply]
All the data should still be in `/data` double check that, if it is not, then I'm afraid the data is either lost, or maybe misplaced.
Else try to search the host for a file called `db.sqlite3` that should be the database file.

**Comment by Section1977** [comment]
it is a fresh install
i tried both sqlite and postgresql

with sqlite i can see the file
but it does not work (master password is not recognized after recreation)


Von meinem iPad gesendet

**Reply by BlackDex** [reply]
If it is a fresh install, with no data, which master password are you expecting to work then?

**Comment by BlackDex** [comment]
I suggest to check and verify if how you start the container is correct.
See: https://github.com/dani-garcia/vaultwarden/wiki/Starting-a-Container#creating-the-container
And: https://github.com/dani-garcia/vaultwarden/wiki/Using-Docker-Compose

**Comment by bokkabonga** [comment]
It might also help if you post the section of your docker-compose in which you handle the mapping of the persistent data folder

**Reply by Section1977** [reply]
thanks a lot for your help
with sqlite I have now success, it works after recreation the container

but with postgresql db the masterpassword settings are still lost after recreation the vaultwarden container
my docker-compose line for using my docker postgresql named postgresql is:

environment:
      - DATABASE_URL='postgresql://bitwarden:bitwarden@postgresql/bitwarden'

Should it work ?

**Reply by Section1977** [reply]
CREATE USER vaultwarden WITH ENCRYPTED PASSWORD 'yourpassword';

is the encryoted password setting creating the db within postgresql essential
i have not encryoted it

**Reply by Section1977** [reply]
?

**Reply by bokkabonga** [reply]
If your Masterpassword is working with sqlite and not with postresql, then i´d guess you didn´t use postresql in the past.
